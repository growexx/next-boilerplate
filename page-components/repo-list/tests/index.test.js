import React from 'react'
import { render, screen, waitFor, fireEvent } from '@testing-library/react'
import { useQuery } from '@tanstack/react-query'

import { RepoList } from '../index'

const mockRepoData = [
  {
    id: 2618963,
    node_id: 'MDEwOlJlcG9zaXRvcnkyNjE4OTYz',
    name: 'AFNetworking',
    full_name: 'js/AFNetworking',
    private: false,
    owner: {
      login: 'js',
      id: 23626
    }
  },
  {
    id: 459905452,
    node_id: 'R_kgDOG2mZrA',
    name: 'analytics-ios-integration-mixpanel',
    full_name: 'js/analytics-ios-integration-mixpanel',
    private: false,
    owner: {
      login: 'js',
      id: 23626
    }
  }
]

// Mock the fetch function and return a mock response
global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve(mockRepoData)
  })
)

// Mocking react-hook-form
jest.mock('react-hook-form', () => ({
  useForm: jest.fn(() => ({
    handleSubmit: jest.fn(),
    formState: {
      errors: {}
    },
    control: {},
    watch: jest.fn()
  }))
}))

// Mocking useQuery
jest.mock('@tanstack/react-query', () => ({
  useQuery: jest.fn(() => ({
    isFetching: false,
    data: mockRepoData,
    isError: false,
    refetch: jest.fn()
  }))
}))

jest.mock('@shared/utils/Fields', () => ({
  Input: jest.fn(() => <input />)
}))

describe('RepoList', () => {
  it('renders the form', () => {
    render(<RepoList />)
    const form = screen.getByTestId('repo-form')
    expect(form).toBeInTheDocument()
  })

  it('displays the list of repositories when the user types', async () => {
    const { container } = render(<RepoList />)
    const input = screen.getByRole('textbox')
    fireEvent.change(input, { target: { value: 'brijesh' } })
    await waitFor(() => expect(useQuery).toHaveBeenCalled())
    expect(container).toMatchSnapshot()
    expect(screen.getByText(`${mockRepoData[0]?.owner.login + '/' + mockRepoData[0]?.name}`)).toBeInTheDocument()
    expect(screen.getByText(`${mockRepoData[1]?.owner.login + '/' + mockRepoData[1]?.name}`)).toBeInTheDocument()
  })
})
