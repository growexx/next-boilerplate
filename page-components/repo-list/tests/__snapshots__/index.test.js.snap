// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoList displays the list of repositories when the user types 1`] = `
<div>
  <div>
    <section
      class="sc-gueYoa lfZlwk"
    >
      <h2>
        Try me!
      </h2>
      <form
        autocomplete="off"
        class="ant-form ant-form-horizontal css-dev-only-do-not-override-1e3x2xa"
        data-testid="REPO_FORM"
      >
        <div
          class="ant-form-item css-dev-only-do-not-override-1e3x2xa"
        >
          <div
            class="ant-row ant-form-item-row css-dev-only-do-not-override-1e3x2xa"
          >
            <div
              class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6 css-dev-only-do-not-override-1e3x2xa"
            >
              <label
                class=""
                for="username"
                title="Show Github repositories by @"
              >
                Show Github repositories by @
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-14 css-dev-only-do-not-override-1e3x2xa"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input />
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
      <div
        class="sc-hLseeU jooqQK"
      >
        <ul
          class="sc-dmqHEX izwLmS"
        >
          <li
            class="sc-fsQiph fHjYAv"
          >
            <div
              class="sc-iAEyYk jFrmgO"
            >
              <div
                class="sc-gLDzan eUiHkE"
              >
                <a
                  class="sc-jTrPJq hHNbiI"
                  target="_blank"
                >
                  js/AFNetworking
                </a>
                <a
                  class="sc-eDDNvR KHfKz"
                  href="undefined/issues"
                  target="_blank"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 14 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </a>
              </div>
            </div>
          </li>
          <li
            class="sc-fsQiph fHjYAv"
          >
            <div
              class="sc-iAEyYk jFrmgO"
            >
              <div
                class="sc-gLDzan eUiHkE"
              >
                <a
                  class="sc-jTrPJq hHNbiI"
                  target="_blank"
                >
                  js/analytics-ios-integration-mixpanel
                </a>
                <a
                  class="sc-eDDNvR KHfKz"
                  href="undefined/issues"
                  target="_blank"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 14 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </a>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </section>
  </div>
</div>
`;
